src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,29 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health = 100, hidden = false) {
+    this.health = health;
+    this.name = name;
+    this.hidden = hidden;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (target instanceof Herbivore && !target.hidden) {
+      target.health -= 50;
+      Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
```

Comment:
The bite method correctly checks if the target is an instance of Herbivore and not hidden before applying damage. After decreasing the health, the Animal.alive array is filtered to remove any animals with health less than or equal to 0, which is an efficient way to handle the removal of dead animals.